### PR TITLE
Koa dev android extend the calendar2 tests with date edge cases

### PR DESCRIPTION
**Pull request by bvitaliyg**
+ [ ] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack-android/blob/main/CONTRIBUTING.md)

Remember to include the following changes:
+ [ ] `UNRELEASED.md`
+ [ ] `README.md`
+ [ ] Tests
+ [ ] Docs (either update [backpack-docs](https://github.com/Skyscanner/backpack-docs) now, or create a follow up ticket)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/main/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

File: Backpack/src/test/java/net/skyscanner/backpack/calendar2/data/CalendarCellsLayoutTests.kt [hunk 1]
```diff
@@ -41,12 +42,10 @@ class CalendarCellsLayoutTests {
...
       verify {
-        assertTrue(state.cells[1] is CalendarCell.Space)
-        assertTrue(state.cells[2] is CalendarCell.Space)
-        assertTrue(state.cells[3] is CalendarCell.Space)
-        assertTrue(state.cells[4] is CalendarCell.Space)
-        assertTrue(state.cells[5] is CalendarCell.Space)
-        assertTrue(state.cells[6] is CalendarCell.Space)
+        for (i in 1..6) {
+          assertTrue(state.cells[i] is CalendarCell.Space)
+          assertEquals(CalendarSettings.Default.range.start.yearMonth(), state.cells[i].yearMonth)
```
Comment (zgeor) [inline Backpack/src/test/java/net/skyscanner/backpack/calendar2/data/CalendarCellsLayoutTests.kt:47]: Does this assertion belong here? Is it related to verifying "there are leading spaces in month" ?

Comment (bvitaliyg) [reply Backpack/src/test/java/net/skyscanner/backpack/calendar2/data/CalendarCellsLayoutTests.kt:47]: Yep, this test verifies there's enough spacings inserted and the spacing have correct format.
